src/main/java/com/jakub/todoSandbox/service/TodoService.java

```diff
@@ -0,0 +1,146 @@
+package com.jakub.todoSandbox.service;
+
+import com.jakub.todoSandbox.model.Step;
+import com.jakub.todoSandbox.model.Todo;
+import org.springframework.stereotype.Service;
+import org.springframework.web.servlet.ModelAndView;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Service
+public class TodoService {
+
+    // Temporary Map for todos - for testing purposes TODO implement postgress database for todos and steps instead of map
+    private Map<Long, Todo> todoList;
+
+    public TodoService(Map<Long, Todo> todoList) {
+        this.todoList = todoList;
+    }
+
+    public ModelAndView isListEmpty() {
+        if(todoList.isEmpty()) {
+            System.out.println("List is empty, returned to index.html");
+            return new ModelAndView("index.html");
+        } else {
+            System.out.println("List is not empty, forwarding to todo-list");
+            return new ModelAndView("/todo-list");
+        }
+    }
+
+    public void createTodo(Todo todo) {
+        todo.setId(todoList.size() + 1L);
+        todoList.put(todo.getId(),todo);
+        System.out.println(validateNameAndDesc(todo.getName(), todo.getDescription()));
+        System.out.println(
+                "New todo was created: \n name: " + todo.getName()
+                + "\n description: " + todo.getDescription()
+                + "\n priority: " + todo.getPriority()
+                + "\n steps: \n" + todo.getNestedSteps().get(0).getName());
+    }
+
+    // Update only the main to-do, not its steps
+    public void updateTodo(Long id, Todo todo) {
+        Todo exists = todoList.get(id);
+        if(exists != null) {
+            exists.setName(todo.getName());
```

Comment:
We should just put new Todo in the map, that's not thread safe and it's is better to use immutable structures in general

Reply:
Yes, but not here, but not-yet-existing InMemoryTodoRepository